examples/ukernels/cpu_brgemm.cpp

```diff
@@ -66,8 +66,17 @@ void brgemm_example() {
     // packing requirements once for multiple objects.
     // Based on this information, specific `ldb` value can be used, since
     // transform has a limited set of values supported.
-    const bool need_pack
-            = brgemm::get_B_pack_type(a_dt, b_dt) == pack_type::pack32;
+    bool need_pack = false;
+    try {
+        need_pack = brgemm::get_B_pack_type(a_dt, b_dt) == pack_type::pack32;
```

Comment:
Are we sure that we want to handle this scenario with exceptions? Another option would be returning `undef` pack type that would indicate that no type is defined for the given configuration.

Reply:
Relying on `undef` likely means an API extension with:
* `undef` meaning something went wrong, re-consider using something other than ukernels;
* `no_pack` meaning packing is not needed;
* other valid values depending on the platform.
, which is probably fine, but not in this PR.

Besides, if going this way, something must be done to ukernel object as well. f32 case won't require packing, will return `no_pack` (with new API) but the kernel is not supported on AVX system, and it won't be a ctor but `finalize()` call.

Reply:
> ndef meaning something went wrong, re-consider using something other than ukernels

For the situations when something went wrong (e.g. invalid arguments) we throw exceptions.

> no_pack meaning packing is not needed;

I couldn't find `no_pack` type, is it something that's going to be introduced? I suggested to rely on `pack_type` based on my understanding that `pack_type` is essentially a layout meaning that anything other than `undef` is OK that is, the operation is defined/supported for the configuration. 

One more thing, how do we expect users who don't want to use exceptions (`--fno-exceptions`) to handle such situations? For primitive descriptors that cannot be created they can use `allow_empty` option but what tools will they have to handle a failed `get_B_pack_type` call?

Reply:
Let me come up with an exception free proposal in a different PR. You raised a good concern.